uisdk/src/test/java/com/karhoo/uisdk/screen/rides/detail/RideDetailPresenterTest.kt

```diff
@@ -565,16 +569,152 @@ class RideDetailPresenterTest {
     }
 
+    /**
+     * Given:   The trip has a service cancellation of type before pickup with a tripStatus different
+     *          than requested or confirmed
+     * Then:    The cancellation text is not shown
+     */
+    @Test
+    fun `When the trip has a service cancellation of type before pickup with a passenger on board trip state, the cancellation text is not shown`() {
+        presenter.checkCancellationSLA(
+                TripStatus.PASSENGER_ON_BOARD,
+                UpcomingRideCardPresenterTest.CANCELLATION_AGREEMENT_BEFORE_PICKUP.freeCancellation,
+                context)
+
+        verify(view, never()).showCancellationText(any())
+        verify(view, never()).setCancellationText(any())
+    }
+
+    /**
+     * Given:   The trip has a service cancellation of type before pickup with a tripStatus equal to confirmed
+     *
+     * Then:    The cancellation text is shown
+     */
+    @Test
+    fun `When the trip has a service cancellation of type before pickup, the cancellation text is shown`() {
+        presenter.checkCancellationSLA(
+                TripStatus.CONFIRMED,
+                UpcomingRideCardPresenterTest.CANCELLATION_AGREEMENT_BEFORE_PICKUP.freeCancellation,
+                context)
+
+        verify(view).showCancellationText(true)
+    }
+
+    /**
+     * Given:   The trip has a service cancellation of type before pickup with a tripStatus equal to confirmed
+     *
+     * Then:    The cancellation text is the correct one
+     */
+    @Test
+    fun `When the trip has a service cancellation of type before pickup, the cancellation text is the correct one`() {
+        presenter.checkCancellationSLA(
+                TripStatus.CONFIRMED,
+                UpcomingRideCardPresenterTest.CANCELLATION_AGREEMENT_BEFORE_PICKUP.freeCancellation,
+                context)
+
+        verify(view).setCancellationText(String.format(BookingQuotesPresenterTest.TEST_CANCELLATION_TEXT, UpcomingRideCardPresenterTest.TEST_TWO_MINUTES))
+    }
+
+    /**
+     * Given:   The trip has a service cancellation of type before driver en route with a tripStatus equal to confirmed
+     *
+     * Then:    The cancellation text is shown
+     */
+    @Test
+    fun `When the trip has a service cancellation of type before driver en route with a confirmed status, the cancellation text is shown`() {
+        presenter.checkCancellationSLA(
+                TripStatus.CONFIRMED,
+                UpcomingRideCardPresenterTest.CANCELLATION_AGREEMENT_BEFORE_DRIVER_EN_ROUTE.freeCancellation,
+                context)
+
+        verify(view).showCancellationText(true)
+    }
+
+    /**
+     * Given:   The trip has a service cancellation of type before driver en route with a tripStatus equal to confirmed
+     *
+     * Then:    The cancellation text is the correct one
+     */
+    @Test
+    fun `When the trip has a service cancellation of type before driver en route, the cancellation text is the correct one`() {
```

Comment:
Can we merge this with the previous one as the visibility and text setting are both part of the 'show'?

Reply:
Merged the tests with the same SUT